plasmapy/particles/particle_collections.py

```diff
@@ -294,6 +294,65 @@ def symbols(self) -> List[str]:
         """A `list` of the symbols of the particles."""
         return self._get_particle_attribute("symbol")
 
+    def average_particle(
+        self,
+        abundances=None,
+        *,
+        use_rms_charge: bool = False,
+        use_rms_mass: bool = False,
+    ) -> Union[CustomParticle, Particle]:
+        """
+        Return a particle with the average mass and charge.
+
+        By default, the mean will be used as the average. If the ``abundances``
+        are provided, then this method will return the weighted mean. If
+        ``use_rms_charge`` or ``use_rms_mass`` is `True`, then return the root
+        mean square of the charge or mass, respectively. If all items in the
+        |ParticleList| are the same, then return that item.
+
+        Parameters
+        ----------
+        abundances : array-like, optional
+            The relative abundances of the particles in the |ParticleList|. Must
+            have the same number of elements as the |ParticleList|.
+
+        use_rms_charge : `bool`, optional, keyword-only
+            If `True`, use the root mean square charge instead of the mean
+            charge. Defaults to `False`.
+
+        use_rms_mass : `bool`, optional, keyword-only
+            If `True`, use the root mean square mass instead of the mean mass.
+            Defaults to `False`.
+
+        Examples
+        --------
+        >>> reactants = ParticleList(["electron", "positron"])
+        >>> reactants.average_particle()
+        CustomParticle(mass=9.109383...e-31 kg, charge=0.0 C)
+        >>> reactants.average_particle(abundances=[1, 0.5])
+        CustomParticle(mass=9.109383...e-31 kg, charge=-5.34058...e-20 C)
+        >>> reactants.average_particle(use_rms_charge=True)
+        CustomParticle(mass=9.109383...e-31 kg, charge=1.6021766...-19 C)
+        >>> protons = ParticleList(["p+", "p+", "p+"])
+        >>> protons.average_particle()
+        Particle("p+")
+        """
+        all_are_particles = all([isinstance(item, Particle) for item in self])
```

Comment:
The methods in `ParticleList` check to make sure that all of the items are a `Particle` or `CustomParticle`, so this should be fine.  I also decided to refactor this part since #1216 makes it possible to use sets for a `ParticleList`.